programs/mpl-core/src/state/collection.rs

```diff
@@ -220,11 +220,16 @@ impl CollectionV1 {
     /// Validate the burn lifecycle event.
     pub fn validate_burn(
         &self,
-        _authority_info: &AccountInfo,
+        authority_info: &AccountInfo,
         _: Option<&Plugin>,
         _: Option<&ExternalPluginAdapter>,
     ) -> Result<ValidationResult, ProgramError> {
-        abstain!()
+        // If the update authority is the one burning the collection, and the collection is empty, then it can be burned.
+        if authority_info.key == &self.update_authority && self.current_size == 0 {
```

Comment:
wdyt about returing a new error if current_size > 0?  Like `CollectionMustBeEmptyToBurn`?  Just an idea, as it results in more specific debug info but at the expense of how `validate_burn` is meant to work as a `CanApprove` method.

Reply:
Yes, this makes sense!